File: ds-grid/src/distributed/systems/grid/simulation/GridSetup.java
package distributed.systems.grid.simulation;

public class GridSetup {
	public String user;
	public String[] schedulers;
	public GridClusterSetup[] clusters;

	public GridSetup(String userId, String[] schedulerIds, GridClusterSetup[] clusterIds) {
		this.user = userId;
		this.schedulers = schedulerIds;
		this.clusters = clusterIds;
	}
}

File: src/models/grid-setup.ts
/**
 * A setup structure used to define every id in a grid.
 */
export interface GridSetup {
    readonly user: string;
    readonly schedulers: string[];
    readonly clusters: GridClusterSetup[];
}

/**
 * Defines the id's in the cluster part of a grid setup.
 */
export interface GridClusterSetup {
    readonly resourceManager: string;
    readonly workers: string[];
}

/**
 * The params used to define the size of a grid.
 */
export interface GridSetupParams {
    readonly schedulers: number;
    readonly clusters: number;
    readonly workers: number;
}
